### PR TITLE
Enclave TCS control

### PR DESCRIPTION
**Pull request by raoulstrackx**
Intel SGXv2 enables the dynamic creation of TCSes. To use this correctly, the enclave needs to be in charge of controlling which TCS will be used for new threads. This PR implements such control for SGXv1 enclaves, as a preliminary step for SGXv2 support.

cc: @jethrogb

### CODE REVIEW

**Comment by raoulstrackx** [comment]
Blocked by https://github.com/fortanix/rust-sgx/pull/300

**Comment by jethrogb** [comment]
Review can be delegated to me

File: src/tools/rustfmt/appveyor.yml [hunk 1]
```diff
@@ -1,55 +1,55 @@
-# This is based on https://github.com/japaric/rust-everywhere/blob/master/appveyor.yml
```
Comment (bjorn3) [inline src/tools/rustfmt/appveyor.yml:1]: Why these changes?

**Comment by Mark-Simulacrum** [comment]
I'm going to go ahead and close this PR, but please feel free to reopen or re-file when the upstream blocking concerns are addressed. Thanks!
